prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -116,6 +118,38 @@ fun Bundle.addProvenanceReference() {
     }
 }
 
+/**
+ * Return true if Bundle contains an ELR in the MessageHeader.
+ *
+ * @return true if has a MesssageHeader that contains an R01 or ORU_R01, otherwise false.
+ */
+fun Bundle.isElr(): Boolean {
+    var isElr = false
+    if (this.type == Bundle.BundleType.MESSAGE && this.entry.isNotEmpty()) {
+        // By rule, the first entry must be a MessageHeader
+        val resource = this.entry[0].resource
+        if (resource is MessageHeader) {
+            val event = resource.event
+            if (event is Coding && ((event.code == "R01") || (event.code == "ORU_R01"))) {
+                isElr = true
+            }
+        }
+    }
+    return isElr
+}
+
+/**
+ * Return RSMessageType based on grouping logic.
+ *
+ * @return RSMessageType of this Bundle.
+ */
+fun Bundle.getRSMessageType(): RSMessageType {
+    when {
```

Comment:
kotlin tip: `when` expressions return values so you can omit your returns in each branch and return the whole expression.

```kotlin
return when {
    isElr() -> RSMessageType.LAB_RESULT
    else -> RSMessageType.UNKNOWN
}
```

Reply:
The Gradle build fails when your suggested change is made.

```
> Task :prime-router:compileKotlin FAILED
e: file:///Users/bill/projects/report-stream/prime-reportstream/prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt:151:1 A 'return' expression required in a function with a block body ('{...}')
```

Reply:
Oops my mistake, apologies.  Made the change as requested.